### PR TITLE
wallet: make the segregation height settable

### PR DESCRIPTION
**Pull request by moneromooo-monero**
via user setting first, then DNS TXT record, hardcoded fallback

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -2594,6 +2597,9 @@ bool wallet2::store_keys(const std::string& keys_file_name, const epee::wipeable
   value2.SetInt(m_key_reuse_mitigation2 ? 1 : 0);
   json.AddMember("key_reuse_mitigation2", value2, json.GetAllocator());
 
+  value2.SetInt(m_segregation_height);
```
Comment (stoffu) [inline src/wallet/wallet2.cpp]: `SetUint`?

File: src/wallet/wallet2.cpp [hunk 2]
```diff
@@ -125,6 +126,7 @@ namespace
     dir /= ".shared-ringdb";
     return dir.string();
   }
+
```
Comment (stoffu) [inline src/wallet/wallet2.cpp]: Intentional?

File: src/simplewallet/simplewallet.cpp [hunk 3]
```diff
@@ -1933,6 +1933,19 @@ bool simple_wallet::set_key_reuse_mitigation2(const std::vector<std::string> &ar
   return true;
 }
 
+bool simple_wallet::set_segregation_height(const std::vector<std::string> &args/* = std::vector<std::string>()*/)
+{
+  const auto pwd_container = get_and_verify_password();
+  if (pwd_container)
+  {
+    parse_bool_and_use(args[1], [&](bool r) {
```
Comment (stoffu) [inline src/simplewallet/simplewallet.cpp]: The expected value isn't bool but uint64_t, so `epee::string_tools::get_xtype_from_string()` should be used instead.

File: src/wallet/wallet2.cpp [hunk 4]
```diff
@@ -10324,6 +10323,58 @@ std::vector<std::pair<uint64_t, uint64_t>> wallet2::estimate_backlog(uint64_t mi
...
+    return TESTNET_SEGREGATION_FORK_HEIGHT;
+  if (m_nettype == STAGENET)
+    return STAGENET_SEGREGATION_FORK_HEIGHT;
+  THROW_WALLET_EXCEPTION_IF(m_nettype != MAINNET, tools::error::wallet_internal_error, "Invalid network type");
+
+  if (m_segregation_height > 0)
+    return m_segregation_height;
+
+  static const bool use_dns = true;
+  if (use_dns)
```
Comment (stoffu) [inline src/wallet/wallet2.cpp]: Does this mean that those who don't want to use the DNS have to compile the code themselves?

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: The user setting takes precedence

Comment (stoffu) [reply src/wallet/wallet2.cpp]: I don't see the reason for the existence of this `if (use_dns) { ... }`.

Comment (moneromooo-monero) [reply src/wallet/wallet2.cpp]: This is because there will likely be a --no-dns flag

Comment (stoffu) [reply src/wallet/wallet2.cpp]: Ah, OK.

**Pull request merged**